lstchain/reco/reconstructor.py

```diff
@@ -0,0 +1,951 @@
+                                    linewidth=6, color='r', linestyle='--',
+                                    label=r'{} $\sigma$ contour'.format(n_sigma))
+            cam_display.axes.legend(loc='best')
+        if init:
+            cam_display.highlight_pixels(self.mask_pixel, color='r')
+
+        if save:
+            cam_display.axes.get_figure().savefig('event/' + ids +
+                                                  '_init' + str(init) + '.png')
+            plt.close()
+        return None if save else cam_display
+
+    def plot_residual(self, image, save=False, ids=''):
+        """
+            Plot the residuals image- spatial_model in the camera after fitting
+
+        Parameters
+        ----------
+        image:
+            Distribution of signal for the event in number of p.e.
+        save: bool
+            Save and close the figure if True, return it otherwise
+        ids: string
+            Can be used to modify the save location
+
+        Returns
+        -------
+        cam_display: `ctapipe.visualization.CameraDisplay`
+            Camera image using matplotlib
+
+        """
+
+        params = self.end_parameters
+
+        rl = 1 + params['rl'] if params['rl'] >= 0 else 1 / (1 - params['rl'])
+        mu = asygaussian2d(np.float32(params['charge']
+                                      * self.geometry.pix_area.to_value(u.m ** 2)),
+                           np.float32(self.geometry.pix_x.value),
+                           np.float32(self.geometry.pix_y.value),
+                           np.float32(params['x_cm']),
+                           np.float32(params['y_cm']),
+                           np.float32(params['wl'] * params['length']),
+                           np.float32(params['length']),
+                           np.float32(params['psi']),
+                           np.float32(rl))
+        residual = image - mu
+
+        cam_display = display_array_camera(residual,
+                                           camera_geometry=self.geometry)
+        if save:
+            cam_display.axes.get_figure().savefig('event/' + ids +
+                                                  '_residuals.png')
+            plt.close()
+        return None if save else cam_display
+
+    def plot_model(self, save=False, ids=''):
+        """
+        Create a CameraDisplay object showing the spatial model fitted to
+        the current event
+
+        Parameters
+        -------
+        save: bool
+            Save and close the figure if True, return it otherwise
+        ids: string
+            Can be used to modify the save location
+
+        Returns
+        -------
+        cam_display: `ctapipe.visualization.CameraDisplay`
+            Camera image using matplotlib
+
+        """
+
+        params = self.end_parameters
+        rl = 1 + params['rl'] if params['rl'] >= 0 else 1 / (1 - params['rl'])
+        mu = asygaussian2d(np.float32(params['charge']
```

Comment:
why using explicitly `np.float32`? is this really needed?

Reply:
I have removed this, and instead modified the compiled function signature to use the type appearing naturally in the code.
What is needed is to have the same type in the function definition (f4 == float32, f8 == float64,...). So assuming that default typing is always the same this local conversion is not needed.